src/leopard_em/pydantic_models/particle_stack.py

```diff
@@ -165,8 +165,20 @@ def construct_image_stack(
             img = load_mrc_image(img_path)
 
             # with reference to center pixel
-            pos_y = self._df.loc[indexes, "pos_y_img"].to_numpy()
-            pos_x = self._df.loc[indexes, "pos_x_img"].to_numpy()
+            pos_y = self._df.loc[
```

Comment:
Would prefer the if/else statement was brought outside of the dataframe indexing. Also should update docstring accordingly to say this method preferentially selects refined columns if they are present